ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -342,9 +344,9 @@ initNetwork registry nodeArgs kernel RunNetworkArgs{..} = do
     nodeToNodeVersionData   = NodeToNodeVersionData { networkMagic   = rnaNetworkMagic }
     nodeToClientVersionData = NodeToClientVersionData { networkMagic = rnaNetworkMagic }
 
-    runLocalServer :: IO ()
-    runLocalServer = do
-      networkState <- newNetworkMutableState
+    runLocalServer :: NetworkMutableState
+                   -> IO ()
+    runLocalServer networkState = do
```

Comment:
@karknu I think it's ok to share the `NetworkMutableState ~ (ConnectionTable, PeerStates)` between local and remote peers.

Reply:
I don't think we should do this. It would introduce an unnecessary synchronization point between services that should have nothing to do with each other.
There is a pre-existing problem with the connection table, we track the connection but we don't know if it was negotiated for NodeToNode or NodeToClient.

Reply:
> All local clients are expected to run on a unix socket - though we can run remote local clients.
Ok, if we want to share the `ConnectionTable` we need to make this explicit.  I wouldn't be worried about to much synchronization: local clients are not expected to die frequently and there's only a few of them running (probably at most 3 at a time).

For now I'll revert this change.